src/os/rtems/src/os-impl-timebase.c

```diff
@@ -390,9 +408,9 @@ int32 OS_TimeBaseCreate_Impl(const OS_object_token_t *token)
         else
         {
             /* will place the task in 'ready for scheduling' state */
-            rtems_sc = rtems_task_start(local->handler_task,                          /*rtems task id*/
-                                        (rtems_task_entry)OS_TimeBase_CallbackThread, /* task entry point */
-                                        (rtems_task_argument)r_name);                 /* passed argument  */
+            rtems_sc = rtems_task_start(local->handler_task,                               /*rtems task id*/
+                                        (rtems_task_entry)OS_TimeBase_CallbackThreadEntry, /* task entry point */
```

Comment:
Can the cast to `rtems_task_entry` be removed here?  Shouldn't be necessary anymore, and may hide errors.

Reply:
I removed it. Without the cast, it used to be a warning (that we could miss if we didn't build with -Werror). Now, with the cast removed and after the updates, it compiles without warnings.